### DISCUSSION TITLE
Youtube series building a computer processor from scratch in Digital

### DISCUSSION THREAD

**Discussion by rj45** [hneemann/Digital#632] [Show and tell]
Hi @hneemann -- I have been looking for a way to share this with you. I hope you don't mind if I also share this with the community.

I have been working on building a computer processor from scratch using Digital. I already have the instruction set designed. I have a way to get GCC working for it, and I plan to port an OS (likely Xinu) to it eventually. But the first few videos I am just incrementally building the simplest thing that could work in more of an educational format.

Here is the YouTube playlist: [Building a CPU from scratch](https://youtube.com/playlist?list=PLilenfQGj6CEG6iZ4TQJ10PI7pCWsy1AO)

**Comment by itviewer** [comment]
why not use risc-v isa?

**Reply by rj45** [reply]
RISC-V is awesome, and there's lots of good reasons to choose it. But I wanted to design my own ISA because it was an interesting problem. So that is what I did.

**Reply by hneemann** [reply]
That would have been my answer as well: 
Although it might be more useful for the general public to implement RISC-V, it's simply much more fun to develop something of your own. :smile:

**Comment by jnoche** [comment]
Thank you for making the videos and sharing them.  I am thinking of using them in my "fundamentals of microprocessors" class that I'm teaching right now.  Aside from the videos, do you have any other material (notes, slides) about the microprocessor you are creating?

**Reply by rj45** [reply]
I am a bit flattered someone would want to use these as part of a course they are teaching. I am okay with that as long as they are supplementary materials, not the primary teaching content. The main reason is that I am afraid that it will go from being a fun hobby in my mind, to being more like a job, and I am worried then that I will lose motivation to continue with the project. Now, that said, I reserve the right to change my mind on this in a few months time. So feel free to ask again in a semester or two and see how I feel.

As for other materials, consider the processor I am building (and its design) to be MIT licensed. If I find the time I can try to get it organized enough to put up on github, and there's roughly a commit per video. But so far this is all I have. I have put a lot of thought into [the instruction set architecture](http://anycpu.org/forum/viewtopic.php?f=23&t=807), and I have some diagrams, but I have no idea if these will be accurate by the time I get to that point many videos into the future.

(Note: The videos are copyrighted and licensed under the standard YouTube license. Only the processor I am building and its design are MIT licensed. This makes screenshots of the videos a bit iffy, I am not sure if they will be MIT or YouTube licensed. But if I put the processor up on github, if you screenshot Digital instead of the YouTube video you should be fine.)

**Reply by jnoche** [reply]
Thank you for your prompt response.  I understand that your processor is a work in progress and a hobby.  I will point out your YouTube channel for my students to watch, but I will search elsewhere for my primary teaching content.  Thanks again.
